src/fdw/trigger.rs

```diff
@@ -274,3 +274,42 @@ fn construct_alter_table_statement(
         column_definitions.join(", ")
     )
 }
+
+#[inline]
+pub fn create_duckdb_relation(
+    table_name: &str,
+    schema_name: &str,
+    table_options: HashMap<String, String>,
+    handler: FdwHandler,
+) -> Result<()> {
+    connection::execute(
+        format!("CREATE SCHEMA IF NOT EXISTS {schema_name}").as_str(),
+        [],
+    )?;
+
+    match handler {
+        FdwHandler::Csv => {
+            connection::create_csv_relation(table_name, schema_name, table_options)?;
+        }
+        FdwHandler::Delta => {
+            connection::create_delta_relation(table_name, schema_name, table_options)?;
+        }
+        FdwHandler::Iceberg => {
+            connection::create_iceberg_relation(table_name, schema_name, table_options)?;
+        }
+        FdwHandler::Parquet => {
+            connection::create_parquet_relation(table_name, schema_name, table_options)?;
+        }
+        FdwHandler::Spatial => {
+            connection::create_spatial_relation(table_name, schema_name, table_options)?;
+        }
+        FdwHandler::Json => {
+            connection::create_json_relation(table_name, schema_name, table_options)?;
+        }
+        _ => {
+            bail!("got unexpected fdw_handler")
+        }
+    };
+
+    Ok(())
+}
```

Comment:
If you think this is a better way to handle creating relations, could you please pull this out into a separate PR and submit it independently? This seems to me irrelevant to the feature at stakes here, which is making things harder to review